homeassistant/components/airnow/__init__.py

```diff
@@ -0,0 +1,184 @@
+"""The AirNow integration."""
+import asyncio
+import datetime
+import logging
+
+from aiohttp.client_exceptions import ClientConnectorError
+from pyairnow import WebServiceAPI
+from pyairnow.conv import aqi_to_concentration
+from pyairnow.errors import AirNowError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_RADIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_API_AQI,
+    ATTR_API_AQI_DESCRIPTION,
+    ATTR_API_AQI_LEVEL,
+    ATTR_API_AQI_PARAM,
+    ATTR_API_CAT_DESCRIPTION,
+    ATTR_API_CAT_LEVEL,
+    ATTR_API_CATEGORY,
+    ATTR_API_PM25,
+    ATTR_API_POLLUTANT,
+    ATTR_API_REPORT_DATE,
+    ATTR_API_REPORT_HOUR,
+    ATTR_API_STATE,
+    ATTR_API_STATION,
+    ATTR_API_STATION_LATITUDE,
+    ATTR_API_STATION_LONGITUDE,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_API_KEY): str,
+                vol.Optional(CONF_LATITUDE): cv.latitude,
+                vol.Optional(CONF_LONGITUDE): cv.longitude,
+                vol.Optional(CONF_RADIUS, default=150): int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["air_quality", "sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the AirNow component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up AirNow from a config entry."""
+    api_key = entry.data[CONF_API_KEY]
+    latitude = entry.data[CONF_LATITUDE]
+    longitude = entry.data[CONF_LONGITUDE]
+    distance = entry.data[CONF_RADIUS]
+
+    # Lat/Long default to Hass Configuration
+    if latitude is None:
+        latitude = hass.configuration.latitude
+    if longitude is None:
+        longitude = hass.configuration.longitude
```

Comment:
Latitude and longitude cannot be `None` because we validate them in config flow.

Reply:
Thanks - updated in 0fdbbff